packages/core/snapshot/snapshotwrite.go

```diff
@@ -0,0 +1,184 @@
+package snapshot
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/iotaledger/goshimmer/packages/core/ledger"
+	"github.com/iotaledger/hive.go/serix"
+)
+
+var delimiter = []byte{';', ';'}
+
+// StreamSnapshotDataTo writes snapshot to a given writer.
+func StreamSnapshotDataTo(
+	writeSeeker io.WriteSeeker,
+	header *ledger.SnapshotHeader,
+	outputProd OutputWithMetadataProducerFunc,
+	epochDiffsProd EpochDiffProducerFunc) error {
+
+	writeFunc := func(name string, value any, offsetsToIncrease ...*int64) error {
+		return writeFunc(writeSeeker, name, value, offsetsToIncrease...)
+	}
+
+	writeOutputWithMetadatasFunc := func(chunks []*ledger.OutputWithMetadata) error {
+		if len(chunks) == 0 {
+			return nil
+		}
+
+		typeSet := new(serix.TypeSettings)
+		data, err := serix.DefaultAPI.Encode(context.Background(), chunks, serix.WithTypeSettings(typeSet.WithLengthPrefixType(serix.LengthPrefixTypeAsUint32)), serix.WithValidation())
+		if err != nil {
+			return err
+		}
+		if err := writeFunc("outputs", append(data, delimiter...)); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	err := writeSnapshotHeader(writeSeeker, header)
+	if err != nil {
+		return err
+	}
+
+	// write outputWithMetadata
+	var outputWithMetadataCounter uint64 = 0
+	var outputChunkCounter int
+	chunksOutputWithMetadata := make([]*ledger.OutputWithMetadata, 0)
+	for {
+		output := outputProd()
+		if output == nil {
+			// write rests of outputWithMetadatas
+			err = writeOutputWithMetadatasFunc(chunksOutputWithMetadata)
+			if err != nil {
+				return err
+			}
+			break
+		}
+
+		outputWithMetadataCounter++
+		chunksOutputWithMetadata = append(chunksOutputWithMetadata, output)
+
+		// put a delimeter every 100 outputs
+		if outputChunkCounter == 100 {
```

Comment:
ok, now I see that we determine the size of chunk here. Maybe lets define this 100 as constant at the top of the file